### PR TITLE
Update GroovyScript & improve material API for groovy

### PR DESCRIPTION
**Pull request by brachy84**
## What
Update GroovyScript to 0.8.0 & improve material API for groovy

## Implementation Details
Just some groovy expansion magic.

## Outcome
Better usage of material api in groovy. Now no imorts are required for basic materials.
The old way is still valid.

## Additional Information
`mods.gregtech.materialEvent` hooks into gregtechs material event without using imports.
`event.materialBuilder(32000, 'flopium')` is added through expansions. The namespace of the resource location is the pack id by default. Idk if this can cause issues if people suddenly use a different namespace.
```groovy
mods.gregtech.materialEvent { event ->
    event.materialBuilder(32000, 'flopium')
        .ingot().fluid()
        .color(0xa87d40)
        .flags("generate_plate", "generate_rod", "generate_gear")
        .components(material('gold'), material('iron') * 3)
        .cableProperties(32, 2, 1)
        .build()
}
```

## Potential Compatibility Issues
None since the old way is still valid.

### CODE REVIEW

**Review by ALongStringOfNumbers — APPROVED** [review]
This seems fine to me (with my limited knowledge), but spotless needs to be run.

**Review by ALongStringOfNumbers — APPROVED** [review]

File: src/main/java/gregtech/integration/groovy/GroovyMaterialBuilderExpansion.java [hunk 1]
```diff
@@ -85,22 +89,29 @@ public static Material.Builder blastTemp(Material.Builder builder, int temp, Str
...
+    public static Material.Builder components(Material.Builder builder, Object... objects) {
+        ObjectArrayList<MaterialStack> materialStacks = new ObjectArrayList<>();
+        for (Object o : objects) {
+            if (o instanceof MaterialStack materialStack) {
+                materialStacks.add(materialStack);
+            } else if (o instanceof Material material) {
+                materialStacks.add(new MaterialStack(material, 1));
+            } else {
+                GroovyLog.get()
+                        .error("Material components must be of type Material or MaterialStack, but was of type {}");
```
Comment (ghzdude) [inline src/main/java/gregtech/integration/groovy/GroovyMaterialBuilderExpansion.java:112]: did you forget to pass in something for formatting the logged message?

**Review by ghzdude — COMMENTED** [review]
just a minor thing, everything else looks fine at a glance

File: src/main/java/gregtech/integration/groovy/GroovyMaterialBuilderExpansion.java (hunk 1 above)
Comment (brachy84) [reply src/main/java/gregtech/integration/groovy/GroovyMaterialBuilderExpansion.java:112]: yes

**Review by ghzdude — APPROVED** [review]

**Pull request merged**